### DISCUSSION TITLE
SEO Friendly URL's from migration and shallow linking

### DISCUSSION THREAD

**Discussion by brideo** [vercel/next.js#15853] [Help]
Hey, 

I am currently migrating a platforms frontend to use Next.JS.

The website is made up of products and categories, the url is made from the product name or the category name.

For example:

Page Type|Page ID|Name|Url|
------------|------------|------------|------------
|category|1|Tablets|`/tablets`|
|category|2|Phones|`/phones`|
|product|1|iPad 4th gen|`/ipad-4th-gen`|

So we do not have to setup a bunch of redirects, in the express server I lookup the route information from an API and change the request url to match the routing used in next:

* `/product/1`
* `/category/1`

This seems to work well, however, I don't know if it's possible to do shallow linking with this method? Say I wanted to link to a different category, is there a way I can do something like this? 

```
    <ul>
      <li>
        <Link href="/category/[id]" as="/category/2" seoUrl="/iphones">
          <a>iPhones</a>
        </Link>
      </li>
    </ul>
```

If this isn't possible, can somebody point me in the direction of the code where I could possibly change the users route? If not I guess I will bite the bullet and redirect the existing urls.

**Comment by jamesmosier** [comment]
You may be able to use either [redirects](https://nextjs.org/docs/api-reference/next.config.js/redirects) or [rewrites](https://nextjs.org/docs/api-reference/next.config.js/rewrites) and then get rid of your custom Express server.

**Reply by brideo** [reply]
I'm trying to avoid from redirects at the moment, if that's not possible I will go that route, thanks for the info.

**Reply by 7iomka** [reply]
> I'm trying to avoid from redirects at the moment, if that's not possible I will go that route, thanks for the info.

Hi, you found any solution?
